### PR TITLE
Restructure the thesis

### PR DESCRIPTION
**Pull request by AleksyBalazinski**

### CODE REVIEW

File: chapters/implementation/implementation.tex [hunk 1]
```diff
@@ -0,0 +1,20 @@
+\chapter{Implementation} \label{ch:implementation}
+The \textit{Hyper} video game is a relatively big project (it comprises around 260 classes), so naturally we have to contain the discussion to only the most important parts of the system.
```
Comment (KarolDenst) [inline chapters/implementation/implementation.tex:2]: Did we ever mention before that we called it Hyper?

File: chapters/implementation/implementation.tex [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+In what follows, we will discuss these areas in more detail.
+We will also provide information on the technologies used in the project and present its core components.
+
+\input{chapters/implementation/sections/technologies_selection/technologies_selection.tex}
+\input{chapters/implementation/sections/game_objects_management/game_objects_management.tex}
+\input{chapters/implementation/sections/terrain/terrain.tex}
+\input{chapters/implementation/sections/chunk_management_system/chunk_management_system.tex}
+\input{chapters/implementation/sections/rendering/rendering.tex}
+\input{chapters/implementation/sections/two_dimensional_graphics/two_dimensional_graphics.tex}
+\todo{I removed the "Bots management" section. I think there wouldn't be much to write about that anyway.}
```
Comment (KarolDenst) [inline chapters/implementation/implementation.tex:20]: I agree. Think we can remove this todo

File: chapters/implementation/sections/two_dimensional_graphics/subsections/menu.tex [hunk 3]
```diff
@@ -29,23 +29,28 @@ \subsection{Menu} \label{subsec:menu}
...
         \caption{Main menu.}
         \label{fig:example_widget}
     \end{subfigure}
 
     \caption{Example widget.}
-\end{figure}
\ No newline at end of file
+\end{figure}
+
+\todo{I copy-pasted that but I don't know if we actually need it}
```
Comment (KarolDenst) [inline chapters/implementation/sections/two_dimensional_graphics/subsections/menu.tex:52]: I think menu is already described enough in other sections

File: chapters/implementation/sections/two_dimensional_graphics/subsections/sprite_renderer.tex [hunk 4]
```diff
@@ -1,4 +1,4 @@
-\section{Sprite Renderer}\label{sprite_renderer}
+\subsection{Sprite renderer} \label{sprite_renderer}
```
Comment (KarolDenst) [inline chapters/implementation/sections/two_dimensional_graphics/subsections/sprite_renderer.tex:1]: Think we should add texttt

File: chapters/implementation/sections/game_objects_management/subsections/controller_classes.tex [hunk 5]
```diff
@@ -1,5 +1,4 @@
-\section{Controllers}
-
+\subsection{\texttt{*Controller} classes}
```
Comment (KarolDenst) [inline chapters/implementation/sections/game_objects_management/subsections/controller_classes.tex:1]: Why the *?

File: chapters/implementation/sections/game_objects_management/subsections/transporter_classes.tex [hunk 6]
```diff
@@ -1,4 +1,4 @@
-\section{Transporter}
+\subsection{\texttt{*Transporter} classes}
```
Comment (KarolDenst) [inline chapters/implementation/sections/game_objects_management/subsections/transporter_classes.tex:1]: why the *?

File: chapters/implementation/sections/game_objects_management/subsections/scene_class.tex [hunk 7]
```diff
@@ -1,18 +1,16 @@
-\section{Scene}
-
+\subsection{\texttt{Scene} class}
 The \texttt{Scene} class is responsible for storing all the objects in the game the player can interact with.
 It is also responsible for disposing of them.
 Here is the full list of objects stored in a \texttt{Scene} instance:
 
 \begin{itemize}
```
Comment (KarolDenst) [inline chapters/implementation/sections/game_objects_management/subsections/scene_class.tex:6]: I don;t like they way enumeration looks both here and in controllers. 10 words take a whole page

File: chapters/implementation/sections/game_objects_management/subsections/controller_classes.tex (hunk 5 above)
Comment (AleksyBalazinski) [reply chapters/implementation/sections/game_objects_management/subsections/controller_classes.tex:1]: To indicate that we have multiple controller classes. But maybe a better name for the section would be "IController interface", then we could write that these are the classes that implement it

File: chapters/implementation/sections/game_objects_management/subsections/transporter_classes.tex (hunk 6 above)
Comment (AleksyBalazinski) [reply chapters/implementation/sections/game_objects_management/subsections/transporter_classes.tex:1]: same as abovve

**Review by KarolDenst — APPROVED** [review]

**Pull request merged**